src/movies.js

```diff
@@ -1,16 +1,79 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array){
+    let directorsArr = array.map((elem)=>{
+    return elem.director
+    })
+    return directorsArr
+    }
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(array){
+    
+    let filterArray = array.filter((elem)=>{
+        return (elem.director==="Steven Spielberg" && elem.genre.indexOf("Drama") != -1)
+      })
+      return filterArray.length
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+function ratesAverage(array){
+    if(array.length === 0){
+        return 0
+    }
+    let average = array.reduce((acc, elem)=>{
+      return acc + elem.rate
+    }, 0)
+    return average / array.length
+  }
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(array){
+    let dramaMovies = array.reduce((acc, elem)=>{
+      if(elem.genre.includes("Drama")){
+        return acc + elem.rate
+      } else{
+        return acc
+      }
+    }, 0)
+    return dramaMovies / dramaMovies.length
```

Comment:
you cannot do dramaMovies.length, as dramaMovies is the sum of rates, so, it is a number, not the array. To get this number you will need to add a new variable before the reduce, and inside the condition, increase the value everytime a new rate is added.